File: credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long cardnum;
    do
    {
        cardnum = get_long("Credit Card Number: ");
    }
    while (cardnum < 1);

    long headnum = cardnum;
    int count = 2;
    while (headnum > 100)
    {
        headnum /= 10;
        count ++;
    }
    long sumnum = cardnum;
    int sum = 0;
    for (int i = 1; i <= count; i++)
    {
        int digit = sumnum % 10;
        if (i % 2 == 0)
        {
            if (digit * 2 > 9)
            {
                sum += (digit * 2 - 9);
            }
            else
            {
                sum += digit * 2;
            }

        }
        else
        {
            sum += digit;
        }
        sumnum /= 10;
    }
    if (sum % 10 == 0)
    {
        if ((count == 13 || count == 16) && (headnum / 10 == 4))
        {
            printf("VISA\n");
        }
        else if ((count == 16) && (headnum > 50 && headnum < 56))
        {
            printf("MASTERCARD\n");
        }
        else if ((count == 15) && (headnum == 34 || headnum == 37))
        {
            printf("AMEX\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: credit.py
while True:
    num = input("Number: ")
    if num.isnumeric() == True:
        break
num = list(map(int, num))


def main():
    clen = len(num)
    first = num[0]
    second = num[1]
    
    if (clen == 15) and (first == 3) and (second == 4 or second == 7):
        if checksum_odd() == 0:
            print("AMEX")
    elif (clen == 16) and (first == 5) and (second <= 5):
        if checksum_even() == 0:
            print("MASTERCARD")
    elif (clen == 13) and (first == 4):
        if checksum_odd() == 0:
            print("VISA")
    elif (clen == 16) and (first == 4):
        if checksum_even() == 0:
            print("VISA")
    else:
        print("INVALID")


def checksum_even():
    sum = 0
    for i in range(0, len(num), 2)[::-1]:
        if num[i] < 5:
            sum += num[i] * 2
        else:
            sum += (((num[i] * 2) // 10) + ((num[i] * 2) % 10))

    for i in range(1, len(num), 2)[::-1]:
        sum += num[i]

    return sum % 10


def checksum_odd():
    sum = 0
    for i in range(1, len(num), 2)[::-1]:
        if num[i] < 5:
            sum += num[i] * 2
        else:
            sum += (((num[i] * 2) // 10) + ((num[i] * 2) % 10))

    for i in range(0, len(num), 2)[::-1]:
        sum += num[i]

    return sum % 10

   
main()